content/3_Silver/Graph_Traversal.mdx

```diff
@@ -1165,3 +1262,81 @@ else:
 ### Graph Two-Coloring Problems
 
 <Problems problems="bip" />
+
+<Quiz>
+	<Quiz.Question>
+		What's the main difference between DFS and BFS?
+
+		<Quiz.Answer>
+			DFS is recursive, while BFS is iterative.
+
+			<Quiz.Explanation>
+				Not really. Both algorithms can be implemented in both recursive and iterative fashion.
+			</Quiz.Explanation>
+		</Quiz.Answer>
+		<Quiz.Answer correct>
+			BFS visits the nodes in order of distance from the start, while DFS follows one path and only backtracks when it needs to.
+
+			<Quiz.Explanation>
+				Correct!
+			</Quiz.Explanation>
+		</Quiz.Answer>
+		<Quiz.Answer>
+			DFS visits the nodes in order of distance from the start, while BFS follows one path and only backtracks when it needs to.
+
+			<Quiz.Explanation>
+				Incorrect. Perhaps you have the terms mixed up?
+			</Quiz.Explanation>
+		</Quiz.Answer>
+	</Quiz.Question>
+	<Quiz.Question>
+		What's the time complexity of these graph traversal algorithms, where $N$ is the number of nodes and $M$ is the number of edges?
+
+		<Quiz.Answer correct>
+			$\mathcal{O}(M+N)$
+
+			<Quiz.Explanation>
+				Correct! We have to iterate over every node and every edge to perform a full search.
+			</Quiz.Explanation>
+		</Quiz.Answer>
+		<Quiz.Answer>
+			$\mathcal{O}(N)$
+
+			<Quiz.Explanation>
+				Sure, we have to iterate over every node, but what about the edges?
+			</Quiz.Explanation>
+		</Quiz.Answer>
+		<Quiz.Answer>
+			$\mathcal{O}(M)$
+
+			<Quiz.Explanation>
+				We do have to loop through all connections, but this expression neglects what the connections actually connect.
+			</Quiz.Explanation>
+		</Quiz.Answer>
+	</Quiz.Question>
+	<Quiz.Question>
+		What collection data structures should we use to implement iterative DFS and BFS?
```

Comment:
im ngl this is kind of a dumb question i dont remember the last time i did iterative dfs

Reply:
do you have a better one

Reply:
i don't remember the last time i did recursive dfs...

Reply:
ask something about connected components

Reply:
yk thinking about it recursive dfs still uses stack in a way
but the connected components idea is cool

Reply:
stacks is a gold topic

Reply:
well it shoudn't be